### PR TITLE
DEV-780 Move file generation to the Validator

### PR DESCRIPTION
**Pull request by michaeldhess**
**High-Level Description**
Update the Validator to handle file generation as well as CSV validations.

**JIRA Ticket**: [DEV-780](https://federal-spending-transparency.atlassian.net/browse/DEV-780)

**Technical Release Notes:**
- Alembic migration to SQSMockQueue to include a file generation agency code
- Move all code related to generating files to the `dataactvalidator`
- Updated D file generation tests

### CODE REVIEW

File: dataactvalidator/validation_handlers/file_generation_manager.py [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+            Args:
+                is_local: True if this is a local installation that will not use AWS
+        """
+        self.is_local = is_local
+        self.sess = GlobalDB.db().session
+
+    def generate_from_job(self, job_id, agency_code):
+        """ Generates a file for a specified job
+            Args:
+                job: upload Job
```
Comment (shinson) [inline dataactvalidator/validation_handlers/file_generation_manager.py]: should be `job_id`

File: tests/unit/dataactvalidator/validation_handlers/test_file_generation_manager.py [hunk 2]
```diff
@@ -0,0 +1,238 @@
...
+    monkeypatch.setattr(file_generation_handler, 'retrieve_job_context_data', Mock(return_value=(sess, new_job)))
+    FileGenerationManager().generate_from_job(new_job.job_id, '123')
+
+    sess.refresh(new_job)
+    assert new_job.original_filename == 'original'
+    assert new_job.from_cached is True
+    assert new_job.job_status_id == sess.query(JobStatus).filter_by(name='finished').one().job_status_id
+
+
+def test_uncache_same_d1_file_fpds_success(monkeypatch, mock_broker_config_paths, database, job_constants):
```
Comment (shinson) [inline tests/unit/dataactvalidator/validation_handlers/test_file_generation_manager.py]: Do we need an uncache same d2 test also?

File: tests/unit/dataactvalidator/validation_handlers/test_file_generation_manager.py [hunk 3]
```diff
@@ -0,0 +1,238 @@
...
+    monkeypatch.setattr(file_generation_handler, 'retrieve_job_context_data', Mock(return_value=(sess, job)))
+    FileGenerationManager().generate_from_job(job.job_id, '123')
+
+    sess.refresh(job)
+    assert job.original_filename != 'original'
+    assert job.from_cached is False
+    assert job.job_status_id == sess.query(JobStatus).filter_by(name='finished').one().job_status_id
+
+
+def test_uncache_new_d1_file_fpds_success(monkeypatch, mock_broker_config_paths, database, job_constants):
```
Comment (shinson) [inline tests/unit/dataactvalidator/validation_handlers/test_file_generation_manager.py]: Do we need an uncache new d2 test?

File: dataactvalidator/app.py [hunk 4]
```diff
@@ -32,40 +35,66 @@ def run_app():
...
+                    # Get the job
+                    job = sess.query(Job).filter_by(job_id=g.job_id).one_or_none()
+                    if job is None:
+                        validation_error_type = ValidationError.jobError
+                        write_file_error(g.job_id, None, validation_error_type)
+                        raise ResponseException('Job ID {} not found in database'.format(g.job_id),
+                                                StatusCode.CLIENT_ERROR, None, validation_error_type)
+
+                    # We have two major functionalities in the Validator: validation and file generation
+                    if not job.file_type or job.file_type.letter_name in ['A', 'B', 'C', 'D2_detached'] or \
```
[the remark this answers is not in the document]
Comment (shinson) [reply dataactvalidator/app.py]: Note: `D2_detached` was renamed to FABS: https://github.com/fedspendingtransparency/data-act-broker-backend/pull/1191/files in lookups.py

**Comment by michaeldhess** [comment]
[run tests]

**Pull request merged**